src/OrchardCore.Modules/OrchardCore.Https/Services/MvcOptionsHttpsConfiguration.cs

```diff
@@ -0,0 +1,27 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+
+namespace OrchardCore.Https.Services
+{
+    public class MvcOptionsHttpsConfiguration : IConfigureOptions<MvcOptions>
```

Comment:
Maybe I missed this in the comments, but could we not just have a middleware do the redirection instead of relying on MVC?